community/values/src/test/java/org/neo4j/values/virtual/ListTest.java

```diff
@@ -198,4 +202,37 @@ public void shouldRecurseAndCoerce()
                 list( "a", list( 'b', list( "c" ) ) ),
                 list( 'a', list( "b", list( 'c' ) ) ) );
     }
+
+    private ListValue[] lists =
+            {VirtualValues.list( Values.longValue( 1L ), Values.longValue( 4L ), Values.longValue( 7L ) ),
+
+                    range( 1L, 8L, 3L ),
+                    VirtualValues.fromArray( Values.longArray( new long[]{1L, 4L, 7L} ) ),
+                    VirtualValues.filter( range( 1L, 100, 1L ), anyValue ->
+                    {
+                        long l = ((LongValue) anyValue).longValue();
+                        return l == 1L || l == 4L || l == 7L;
+                    } ),
+                    VirtualValues.slice( list( -2L, 1L, 4L, 7L, 10L ), 1, 4 ),
+                    VirtualValues.drop( list( -2L, 1L, 4L, 7L ), 1 ),
+                    VirtualValues.take( list( 1L, 4L, 7L, 10L, 13L ), 3 ),
+                    VirtualValues.transform( list( 0L, 3L, 6L ),
+                            anyValue -> Values.longValue( ((LongValue) anyValue).longValue() + 1L ) ),
+                    VirtualValues.reverse( list( 7L, 4L, 1L ) )};
```

Comment:
yes, we have taken the static approach for other stuff so figured I'd stick with that. Maybe we can discuss this and change in a future PR if that makes more sense.